ntoskrnl/io/iomgr/deviface.c

```diff
@@ -30,185 +31,437 @@ static PWCHAR BaseKeyString = L"\\Registry\\Machine\\System\\CurrentControlSet\\
 
 static
 NTSTATUS
-OpenRegistryHandlesFromSymbolicLink(IN PUNICODE_STRING SymbolicLinkName,
-                                    IN ACCESS_MASK DesiredAccess,
-                                    IN OPTIONAL PHANDLE GuidKey,
-                                    IN OPTIONAL PHANDLE DeviceKey,
-                                    IN OPTIONAL PHANDLE InstanceKey)
+IopBuildSymbolicLink(_In_ PUNICODE_STRING PrefixString,
+                     _In_ PUNICODE_STRING DeviceString,
+                     _In_ PUNICODE_STRING GuidString,
+                     _In_opt_ PUNICODE_STRING ReferenceString,
+                     _Out_ PUNICODE_STRING SymbolicLinkName)
```

Comment:
Maybe you could use the "new" presentation of prototypes, here and in the other added/modified functions:
```suggestion
IopBuildSymbolicLink(
    _In_ PCUNICODE_STRING PrefixString,
    _In_ PCUNICODE_STRING DeviceString,
    _In_ PCUNICODE_STRING GuidString,
    _In_opt_ PCUNICODE_STRING ReferenceString,
    _Out_ PUNICODE_STRING SymbolicLinkName)
```
and also note the `PUNICODE_STRING` --> `PCUNICODE_STRING` suggestions as these input-only strings are not modified.

Reply:
Thanks. Updated all prototypes of the functions according to your comment. :slightly_smiling_face: